numpy/core/code_generators/ufunc_docstrings.py

```diff
@@ -2541,6 +2541,128 @@ def add_newdoc(place, name, doc):
 
     """)
 
+add_newdoc('numpy.core.umath', 'matmul',
+    """
+    Matrix product of two arrays.
+
+    Parameters
+    ----------
+    x1, x2 : array_like
+        Input arrays.
+    out : ndarray, optional
+        Output argument. This must have the exact kind that would be returned
+        if it was not used. In particular, it must be C-contiguous and have
+        both the right type and the right dtype. This is a performance feature.
+        Therefore, if these conditions are not met, an exception is raised,
+        instead of attempting to be flexible.
+    **kwargs
+        For other keyword-only arguments, see the
+        :ref:`ufunc docs <ufuncs.kwargs>`.
+
+
+    Returns
+    -------
+    y : ndarray
+        The matrix product of the inputs
+        $OUT_SCALAR_2
```

Comment:
I think this expands to "This is a scalar if both `x1` and `x2` are scalars." - which is not true - it is a scalar only when `x1` and `x2` are 1-d vectors. I think you'll need to define a new variable...

Reply:
fixed.